File: RedSheet.DbEntities/Models/vProjectCulturePlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectCulturePlans", Schema = "dbo")]
	[Serializable]
	public partial class vProjectCulturePlan  
	{

	public Nullable<int> CulturePlanCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public Nullable<int> CulturePlanId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int ProjectCulturePlanId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string ProjectCulturePlanValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 

	public vProjectCulturePlan()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-culture-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectCulturePlan {
    constructor(vProjectCulturePlan?: vProjectCulturePlan )  {
        let properties = [ "culturePlanCategoryId", "culturePlanId", "projectCulturePlanId", "projectCulturePlanValue", "projectModuleId",];
        for (let property of properties)
            if (vProjectCulturePlan && vProjectCulturePlan[property])
                this[property] = vProjectCulturePlan[property];
    }
 
	culturePlanCategoryId : number =   undefined;
 
	culturePlanId : number =   0 ;
 
    @range(1,2147483647)
	projectCulturePlanId : number =   undefined;
 
    @required()
    @maxLength(1000)
	projectCulturePlanValue : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;


}
